src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -4041,10 +4041,12 @@ bool PhaseIdealLoop::intrinsify_fill(IdealLoopTree* lpt) {
     index = new LShiftXNode(index, shift->in(2));
     _igvn.register_new_node_with_optimizer(index);
   }
-  index = new AddPNode(base, base, index);
-  _igvn.register_new_node_with_optimizer(index);
-  Node* from = new AddPNode(base, index, offset);
+  Node* from = new AddPNode(base, base, index);
   _igvn.register_new_node_with_optimizer(from);
+  if (offset != NULL) {
```

Comment:
A comment here that this happens with unsafe would be useful.

Reply:
Comments are added. Thanks!